### PR TITLE
Rchain 2831/casper/complex bonding scenarios

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
Test showing assertion error in new SafetyOracle.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
RCHAIN-2831

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

File: casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala [hunk 1]
```diff
@@ -1,36 +1,72 @@
...
+import coop.rchain.shared.AtomicMonadState
+import monix.execution.atomic.AtomicAny
 
-class TransportLayerTestImpl[F[_]: Monad](
-    identity: PeerNode,
-    val msgQueues: Map[PeerNode, Ref[F, mutable.Queue[Protocol]]]
+import scala.collection.immutable.Queue
+
+object TestNetwork {
+  type Topology          = Map[PeerNode, Queue[Protocol]]
```
Comment (KentShikama) [inline casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: Why topology? It feels like indexed queues.

**Pull request merged**